src/main/java/Doot.java

```diff
@@ -0,0 +1,143 @@
+        }
+        System.out.print(DIVIDER + "Bye. Hope to see you again soon!" + "\n" + DIVIDER);
+        scanner.close();
+    }
+
+    public static void findCommand(String command) {
+        String wordDigit = "\\d{1,3}$";
+        String deadlineMatch = "(\\w+ )*/by .+";
+        String eventMatch = "(\\w+ )*/from (\\w+ )*/to .+";
+        boolean containsSpace = command.contains(" ");
+        String cmd = containsSpace ? command.substring(0, command.indexOf(" ")) : command;
+        String args = containsSpace ? command.substring(command.indexOf(" ") + 1) : "";
+        if (args.matches(wordDigit)) {
+            handleWordDigit(cmd, args);
+        } else if (args.matches(eventMatch)) {
+            handleEvent(cmd, args);
+        } else if (args.matches(deadlineMatch)) {
+            handleDeadline(cmd, args);
+        } else {
+            handleDefault(cmd, args);
+        }
+    }
+
+    private static void handleWordDigit(String command, String args) {
+        int digit = Integer.parseInt(args);
+        switch (command) {
+            case "mark":
+                markTask(digit);
+                break;
+            case "unmark":
+                unmarkTask(digit);
+                break;
+            default:
+                addToList(command);
+                break;
+        }
+    }
+
+    private static void handleDeadline(String command, String args) {
+        String[] parts = args.split(" /by ");
+        String wordOne = parts[0];
+        String wordTwo = parts[1];
+        switch (command) {
+            case "deadline":
+                makeDeadline(wordOne, wordTwo);
+                break;
+            default:
+                addToList(command + args);
+                break;
+        }
+    }
+
+    private static void handleEvent(String command, String args) {
+        String[] parts = args.split(" /from ");
+        String wordOne = parts[0];
+        String wordTwo = parts[1];
+        parts = wordTwo.split(" /to ");
+        wordTwo = parts[0];
+        String wordThree = parts[1];
+        switch (command) {
+            case "event":
+                makeEvent(wordOne, wordTwo, wordThree);
+                break;
+            default:
+                addToList(command + args);
+                break;
+        }
+    }
+
+    private static void handleDefault(String command, String args) {
+        switch (command){
+            case "todo":
+                makeToDo(args);
+                break;
+            case "list":
+                printList();
+                break;
+            default:
+                addToList(command + args);
+        }
+    }
+
+    public static void makeDeadline(String description, String by){
+        taskList[taskIdx] = new Deadline(description, by);
+        taskIdx++;
+        System.out.print(DIVIDER + "Got it. I've added this task:\n" + taskList[taskIdx-1].toString() + "\n" + "Now you have " + taskIdx + " tasks in the list.\n" + DIVIDER);
```

Comment:
Can consider putting all the print statements into one print class and use toString method for your original class